src/idlcxx/src/streamers.c

```diff
@@ -665,8 +879,10 @@ generate_streamers(const idl_pstate_t* pstate, struct generator *gen)
   sources[0] = pstate->sources->path->name;
   visitor.sources = sources;
 
-  if ((ret = idl_visit(pstate, pstate->root, &visitor, &streams)) == IDL_RETCODE_OK)
-    ret = flush(gen, &streams);
+  idl_retcode_t ret = IDL_RETCODE_OK;
+  if (((ret = idl_visit(pstate, pstate->root, &visitor, &streams)) != IDL_RETCODE_OK) ||
+      ((ret = flush(gen, &streams)) != IDL_RETCODE_OK))
```

Comment:
You may have to cleanup the streams here. It may have failed half-way through, probably best to add an error label or something which frees and returns `ret`.

Reply:
Check.

I will go over the paths and see when and where to clean up.